mmcv/ops/deform_conv.py

```diff
@@ -70,8 +70,14 @@ def forward(ctx,
         ctx.deform_groups = deform_groups
         ctx.im2col_step = im2col_step
 
-        # until the code is modified for torch.cuda.amp.autocast,
-        # we need to cast weight to avoid type mismatch in fp16 training
+        # When pytorch version >= 1.6.0, amp is adopted for fp16 mode;
+        # amp won't cast the type of model (float32), but "offset" is cast
+        # to float16 by nn.Conv2d automatically, leading to the type
+        # mismatch with input (when it is float32) or weight.
+        # The flag for whether to use fp16 or amp is the type of "offset",
+        # we cast weight and input to temporarily support fp16 and amp
+        # whatever the pytorch version is.
+        input = input.to(offset.dtype)
         weight = weight.type_as(input)
```

Comment:
We can use .to() or .type_as() for all cases.  Please unify the usages.

Reply:
@ZwwWayne Done.